includes/Order/Hooks.php

```diff
@@ -292,7 +292,9 @@ public function ensure_vendor_coupon( $valid, $coupon ) {
         foreach ( WC()->cart->get_cart() as $item ) {
             $product_id = $item['data']->get_id();
 
-            $available_vendors[]  = (int) get_post_field( 'post_author', $product_id );
+            $vendor = dokan_get_vendor_by_product( $product_id );
```

Comment:
Pass 2nd parameter `true` then the function will be returned directly `vendor_id`
```
$vendor_id                  = dokan_get_vendor_by_product( $product_id, true );
$available_vendors[]  = $vendor_id ? $vendor_id : 0;
```